docs/api.md

```diff
@@ -53,13 +53,94 @@ In our experience, any system that is successful needs to grow and change as new
+     - a feature has had an API review, though the beta period is still likely to turn up API issues not
+       found in review.
+     - the project is committed to graduating the feature to stable (typically within 3 months, but sometimes longer)
+     - has been tested to high standards and is considered safe to enable in a production cluster.
+   - The object schema may change in a later software release.  When this happens, an automatic
+     upgrade of a cluster with beta objects may not be possible.  Instead, it may be necessary
+     to delete/modify/re-create objects in order to upgrade a cluster to the new schema.  When this happens,
+     for a beta feature, we will provide documentation on how to do this.
+   - Releases will support two consecutive versions (e.g. `v1beta1` and `v1beta2`; or `v1beta2` and `v1`)
+     for at least one minor release cycle (typically 3 months) so that users have enough time
+     to upgrade and migrate objects.
+   - TBD: do beta features enable by default?
+   - TBD: do we commit that released beta features work on GKE?
+     to allow for migrating from one version to the next.  (Releases are typically 3 months apart.)
+- Stable level:
+   - The version name is `vX` where `X` is an integer.
+   - Stable versions will be supported for a long time.
+
+## API groups
+
+To make it easier to extend the Kubernetes API, we are in the process of implementing [*API
+groups*](proposals/api-groups.md).  These are simply different interfaces to read and/or modify the
+same underlying resources.  The API group is specified in a REST path and in the `apiVersion` field
+of a serialized object.
+
+Currently there are two API groups in use:
+
+1. the "core" group, which is at REST path `/api/v1` and is not specified as part of the `apiVersion` field, e.g.
+   `apiVersion: v1`.
+1. the "experimental" group, which is at REST path `/apis/experimental/$VERSION`, and which uses
+  `apiVersion: experimental/$VERSION` (e.g. currently `apiVersion: experimental/v1alpha1`).
+  The name `experimental` indicates that API groups themselves are currently under development.  The actual
+  stability of the version of objects within that group is determined by its [API
+  version](#api-versioning)
+
+In the future we expect that there will be more API groups, all at REST path `/apis/$API_GROUP` and
+using `apiVersion: $API_GROUP/$VERSION`.  We expect that there will be a way for (third parties to
+create their own API groups](design/extending-api.md), and to avoid naming collisions.
+
+Not all Kubernetes clusters will support all API groups.  However, we intend that if an API server
+supports a group, it supports all the Kinds in the group, and all of them at the same version(s).
+
+## Enabling `experimental/v1alpha1`
+
+Enable `experimental/v1alpha1` objects by adding the following flags to your API server:
+
+  - `--runtime-config=experimental/v1alpha1=true`
+  - `--enable-experimental=true`
```

Comment:
@nikhiljindal should I document how to turn off individual kinds using runtime-config?